### PR TITLE
Add threat model docs

### PR DESCRIPTION
**Pull request by portersrc**
Addresses issues https://github.com/confidential-containers/confidentialcontainers.org/issues/24 and https://github.com/confidential-containers/confidentialcontainers.org/issues/25

Also pulls in the threats-overview file:
https://github.com/confidential-containers/confidential-containers/blob/main/threats_overview.md

PR to remove these docs from confidential-containers: https://github.com/confidential-containers/confidential-containers/pull/190

### CODE REVIEW

**Comment by portersrc** [comment]
Hi @surajssd,

I threw the threat docs under Overview; maybe double-check that that seems reasonable.

Also, I tried two different ways of displaying the personas_model.svg (see [here](https://github.com/confidential-containers/confidential-containers/blob/main/trust_model_personas.md) for how it used to look). With the svg file in `static/`, both of these lines work on a local `docker compose up` test:
```Persona model```
``````
But neither are showing in the netlify preview. Any advice welcome.

**Comment by surajssd** [comment]
@portersrc the svg file is invalid.

**Comment by portersrc** [comment]
Yuck. Good catch. It seems to be displaying properly now. Thanks!

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+firmware used to provide compute, network and storage to the Cloud Native solution.
+They are responsible for availability of infrastructure used by the cloud native environment.
+- Have access to the physical hardware.
+- Have access to the processes involved in the deployment of compute/storage/memory used by any
+  orchestration components and by the workload.
+- Have control over TEE hardware availability/type.
+- Responsibility for applying firmware updates to infrastructure including the TEE Technology.
+
+Example : Cloud Service Provider (CSP), Infrastructure as a Service (IaaS), Site Reliability Engineer, etc.
+(SRE)
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:58]:
```suggestion
Examples: Cloud Service Provider (CSP), Site Reliability Engineer (SRE), etc.
```

IaaS is  product category, not a subject

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+- Control availability of service.
+- Control webhooks and deployment of workloads.
+- Control availability of cluster resources (data/networking/storage) and cluster
+  services (Logging/Monitoring/Load Balancing) for the workloads.
+- Control the deployment of runtime artifacts initially required by the TEE during
+  initialisation.
+ These boot images once initialised will receive the workload.
+
+Example : A kubernetes administrator responsible for deploying pods to a cluster and
+maintaining the cluster.
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:74]:
```suggestion
Example: A Kubernetes administrator responsible for deploying pods to a cluster and
maintaining the cluster.
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+They are responsible for deploying a solution into a particular cloud native environment and
+managing the orchestration environment.
+For managed cluster this would also include the administration of the cluster control plane.
+- Control availability of service.
+- Control webhooks and deployment of workloads.
+- Control availability of cluster resources (data/networking/storage) and cluster
+  services (Logging/Monitoring/Load Balancing) for the workloads.
+- Control the deployment of runtime artifacts initially required by the TEE during
+  initialisation.
+ These boot images once initialised will receive the workload.
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:71]: boot images is suddenly introduced as a term.  3x "initially/ization"

```suggestion
- Control the deployment of runtime artifacts required by the TEE during
  initialisation, before hosting the confidential workload.
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+Example : A kubernetes administrator responsible for deploying pods to a cluster and
+maintaining the cluster.
+
+### Workload Provider
+This persona designs and creates the orchestration objects comprising the solution (e.g.
+kubernetes pod descriptions etc). These objects reference containers published by Container Image Providers.
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:78]:
```suggestion
Kubernetes Pod spec, etc.). These objects reference containers published by Container Image Providers.
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+In some cases the Workload and Container Image Providers may be the same entity.
+The solution defined is intended to provide the Application or Workload which in turn provides
+value to the data owners (customers and clients).
+The workload provider and data owner could be part of same company/organisation but
+following the least privilege principle the workload provider should not be able to view or
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:83]: If the capitalize Personas and Roles (i think we should) we should do it consistently

```suggestion
value to the Data Owners (customers and clients).
The Workload Provider and Data Owner could be part of same company/organisation but
following the least privilege principle the Workload Provider should not be able to view or
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+manipulate end user data without informed consent.
+- Need to prove to customer aspects of compliance.
+- Defines what the solution requires in order to run and maintain compliance (resources, utility
+  containers/services,  storage).
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:87]:
```suggestion
  containers/services, storage).
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+- Chooses the method of verifying the container images (from those supported by Container Image
+  Provider) and obtains artifacts needed to allow verification to be completed within
+  the TEE.
+- Provide the boot images initially required by the TEE during
+  initialisation or designates a trusted party to do so.
+- Provide the attestation verification service, or designate a trusted party to provide the
+  attestation verification service.
+
+Example : 3rd party software vendor, cloud solution provider
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:96]: CSP abbrev introduced above

```suggestion
Examples: 3rd party software vendors, CSP
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+  controlled by a Key Broker Service).
+- Provides other required verification artifacts (secure channel may be considered).
+- Protects the keys used to sign or encrypt the container images.
+
+It is recognised that hybrid options exist surrounding workload provider and container provider.
+For example the workload provider may choose to protect their supply chain by
+signing/encrypting their own container images after following the build patterns already
+established by the container image provider.
+
+Example : Sidecar Container Provider
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:123]: that doesn't seem to be a tangible example for Container Image Provider

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+Outside the TEE we distinguish between the Orchestration Operator and the Infrastructure
+Operator due to nature of how they can impact the TEE and the concerns of Workload Provider and
+Data Owner. Direct threats exist from the Orchestration Operator as some orchestration actions
+must be permitted to cross the TEE boundary otherwise orchestration cannot occur. A key goal is to
+*deprivilege orchestration* and restrict the
+Orchestration Operators privileges across the boundary. However indirect threats exist
+from the Infrastructure Operator who would not be permitted to exercise orchestration APIs but
+could exploit the low level hardware or firmware capabilities to access or impact the  contents
+of a TEE.
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:169]:
```suggestion
Orchestration Operators privileges across the boundary. However indirect threats exist
from the Infrastructure Operator who would not be permitted to exercise orchestration APIs but
could exploit the low-level hardware or firmware capabilities to access or impact the contents
of a TEE.
```

File: content/en/docs/overview/trust-model/trust_model_personas.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+
+### Orchestration Operator vs. Infrastructure Operator
+
+Outside the TEE we distinguish between the Orchestration Operator and the Infrastructure
+Operator due to nature of how they can impact the TEE and the concerns of Workload Provider and
+Data Owner. Direct threats exist from the Orchestration Operator as some orchestration actions
+must be permitted to cross the TEE boundary otherwise orchestration cannot occur. A key goal is to
+*deprivilege orchestration* and restrict the
+Orchestration Operators privileges across the boundary. However indirect threats exist
+from the Infrastructure Operator who would not be permitted to exercise orchestration APIs but
```
Comment (mkulke) [inline content/en/docs/overview/trust-model/trust_model_personas.md:167]: > Infrastructure Operator who would not be permitted to exercise orchestration APIs

is this assumption warranted? isn't the infra operator inherently privileged to access orchestration APIs? (like edit values in etcd)

**Review by mkulke — COMMENTED** [review]
I wonder if the `Container Image Provider` persona is tied too much to the assumption that users will deploy encrypted/signed container images and should be revisited. There is no mention of secure key-release at runtime, for example. Maybe it can be generalized into a "Secret Provider" persona, i.e. someone who manages secrets?

File: content/en/docs/overview/trust-model/trust_model.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+In order to describe and understand particular threats we need to establish trust boundaries and
+trust models relating to the key aspects, components and actors involved in Cloud Native
+Confidential Computing. We explore trust using different orthogonal ways of considering cloud
+native approaches when they use an underlying TEE technology and
+identifying where there may be considerations to preserve the value of using a TEE.
+
+### Trust Model Considerations
+- [Personas](./trust_model_personas.md)
+
+Further documentation will highlight specific [threat vectors](./threats_overview.md) in detail,
```
[the remark this answers is not in the document]
Comment (portersrc) [reply content/en/docs/overview/trust-model/trust_model.md]: good catch. maybe similar issues here https://deploy-preview-27--frolicking-manatee-96c0c8.netlify.app/docs/kata-containers/kata-confidential-containers/

File: content/en/docs/overview/trust-model/trust_model_personas.md (hunk 8 above)
Comment (portersrc) [reply content/en/docs/overview/trust-model/trust_model_personas.md:123]: Are you looking for an adjective or description that makes this more concrete (i.e. "tangible"), e.g. "Monitoring Sidecar Container Provider"?

File: content/en/docs/overview/trust-model/trust_model_personas.md (hunk 10 above)
Comment (portersrc) [reply content/en/docs/overview/trust-model/trust_model_personas.md:167]: I think the distinction here is also weak. I wouldn't change it. If you want something else, though, I might suggest:
> However, threats exist from the Infrastructure Operator who could indirectly exercise orchestration APIs or exploit low-level hardware or firmware capabilities to access or impact the contents of a TEE.

File: content/en/docs/overview/trust-model/trust_model.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (portersrc) [reply content/en/docs/overview/trust-model/trust_model.md]: Seems to have an issue for it already. Resolving. https://github.com/confidential-containers/confidentialcontainers.org/issues/5

**Comment by portersrc** [comment]
> I wonder if the Container Image Provider persona is tied too much to the assumption that users will deploy encrypted/signed container images and should be revisited. There is no mention of secure key-release at runtime, for example. Maybe it can be generalized into a "Secret Provider" persona, i.e. someone who manages secrets?

I don't think you're wrong here. That also would affect the personas diagram. I'm OK with updating that or leaving as-is. @magowan any thoughts on this (because I think you supplied a lot of the initial ideas [here](https://github.com/confidential-containers/confidential-containers/commit/c6af54631aa963dd437cba2028dc99b61977d7e0))?

File: content/en/docs/overview/trust-model/trust_model_personas.md (hunk 8 above)
Comment (mkulke) [reply content/en/docs/overview/trust-model/trust_model_personas.md:122]: ah, what I meant is: this seems like a category that would itself need to be exemplified, i.e. I don't really know what a "(Monitoring) Sidecar Container Provider" would be. is that istio, linkerd, dockerhub or whoever owns the supplychain for a sidecar container?

**Review by mkulke — COMMENTED** [review]
Maybe we don't need to address the conceptual questions (infrastructure vs orchestrator op & container image provider vs secret provider) in the scope of this PR, but open issues in which we discuss those separately. We probably want to get the input from more people.

The prose linter seems to object to some words, you can add them to `./styles/config/vocabularies/coco/accept.txt` to fix the errors.

File: content/en/docs/overview/trust-model/trust_model_personas.md (hunk 8 above)
Comment (portersrc) [reply content/en/docs/overview/trust-model/trust_model_personas.md:122]: Oh, I see. Your examples make sense, and I can put istio here.

**Review by mkulke — APPROVED** [review]
LGTM

I'll add issues about the generic questions to the CoCo Threat model

**Comment by surajssd** [comment]
@portersrc @mkulke can we merge this? or we are waiting on something?

**Comment by portersrc** [comment]
I think we're OK to merge. @mkulke indeed raised follow-up issues:
https://github.com/confidential-containers/confidential-containers/issues/194
https://github.com/confidential-containers/confidential-containers/issues/195

I will bump the [PR in the main repo](https://github.com/confidential-containers/confidential-containers/pull/190) if/when you merge this.

**Pull request merged**